internal/pkg/cli/svc_deploy.go

```diff
@@ -310,252 +284,92 @@ func (o *deploySvcOpts) askEnvName() error {
 }
 
 func (o *deploySvcOpts) configureClients() error {
-	defaultSessEnvRegion, err := o.sessProvider.DefaultWithRegion(o.targetEnvironment.Region)
-	if err != nil {
-		return fmt.Errorf("create ECR session with region %s: %w", o.targetEnvironment.Region, err)
-	}
-
-	envSession, err := o.sessProvider.FromRole(o.targetEnvironment.ManagerRoleARN, o.targetEnvironment.Region)
-	if err != nil {
-		return fmt.Errorf("assuming environment manager role: %w", err)
-	}
-
-	d, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
-		App:         o.appName,
-		Env:         o.envName,
-		ConfigStore: o.store,
-		DeployStore: o.deployStore,
-	})
-	if err != nil {
-		return fmt.Errorf("create describer for environment %s in application %s: %w", o.envName, o.appName, err)
-	}
-	o.envDescriber = d
-
-	// ECR client against tools account profile AND target environment region.
-	repoName := fmt.Sprintf("%s/%s", o.appName, o.name)
-	registry := ecr.New(defaultSessEnvRegion)
-	o.imageBuilderPusher, err = repository.New(repoName, registry)
+	o.imageTag = imageTagFromGit(o.cmd, o.imageTag) // Best effort assign git tag.
```

Comment:
A question! How do we decide what clients should be configured here versus `cli/deploy`?

Reply:
It's just intuitively since we get the value from the tag and assign default value if they don't specify the flag, I feel it should be still in the `cli` package as it is input from users like `app`, `env`, and `svc`.

Reply:
ohhh wrong place to add the comment! It was more like a general questions, e.g. what about `appVersionGetter` and `appRegionalResources`? Are they configured in `cli` for the same reason?

Reply:
ooh good point i think i just missed them